process/block/shardblock.go

```diff
@@ -285,7 +286,17 @@ func (sp *shardProcessor) checkEpochCorrectness(
 
 	if sp.endOfEpochTrigger.IsEndOfEpoch() &&
 		headerHandler.GetRound() > sp.endOfEpochTrigger.EpochStartRound()+process.MetaBlockFinality &&
-		headerHandler.GetEpoch() != currentBlockHeader.GetEpoch()+1 {
+		headerHandler.GetEpoch() != sp.endOfEpochTrigger.Epoch() {
```

Comment:
can you extract the conditions into variables?

Reply:
done